File: SampleApps/ReceiverDemo/RouteHandlers/DPLInventoryRouteHandler.m
#import "DPLInventoryRouteHandler.h"

@implementation DPLInventoryRouteHandler

- (BOOL)preferAsynchronous {
    return YES;
}

- (void)targetViewControllerWithCompletion:(void (^)(UIViewController<DPLTargetViewController> *))completionHandler {
    UIStoryboard *storyboard = [UIApplication sharedApplication].keyWindow.rootViewController.storyboard;
    
    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
        UIViewController<DPLTargetViewController> *vc = [storyboard instantiateViewControllerWithIdentifier:@"inventory"];
        completionHandler(vc);
    });

}

@end

File: SampleApps/ReceiverDemoSwift/RouteHandlers/DPLInventoryRouteHandler.swift
import UIKit

public class DPLInventoryRouteHandler: DPLRouteHandler {
    public override func preferAsynchronous() -> Bool {
        return true
    }
    
    public override func targetViewControllerWithCompletion(completionHandler: ((UIViewController!) -> Void)!) {
        guard let storyboard = UIApplication.sharedApplication().keyWindow?.rootViewController?.storyboard else {
            completionHandler(nil)
            return
        }
        
        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (Int64)(2 * NSEC_PER_SEC)), dispatch_get_main_queue()) {
            let vc = storyboard.instantiateViewControllerWithIdentifier("inventory") as! DPLInventoryTableViewController
            completionHandler(vc)
        }
    }
}
